File: tools/sms/src/test/java/top/bettercode/summer/tools/sms/aliyun/AliSmsTemplateTest.java
package top.bettercode.summer.tools.sms.aliyun;


import kotlin.Pair;
import kotlin.collections.MapsKt;
import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import top.bettercode.summer.test.BaseTest;

/**
 * @author Peter Wu
 */
@Disabled
class AliSmsTemplateTest extends BaseTest {

  @Autowired
  AliSmsTemplate aliSmsTemplate;
  @Value("${mobile}")
  String mobile;

  @Disabled
  @Test
  void sendSms() {
    aliSmsTemplate.sendSms("SMS_126359654", new AliSmsReq(mobile, "云图提货", MapsKt.mapOf(
        new Pair<>("code", "123456"))));
  }

  @Test
  void querySendReport() {
    aliSmsTemplate.querySendReport(mobile);
  }

  @Test
  void querySmsSignList() {
    aliSmsTemplate.querySmsSignList();
  }

  @Test
  void querySmsTemplateList() {
    aliSmsTemplate.querySmsTemplateList();
  }

}

File: tools/sms/src/test/kotlin/top/bettercode/summer/tools/sms/aliyun/AliSmsTemplateTest.kt
package top.bettercode.summer.tools.sms.aliyun

import org.junit.jupiter.api.Disabled
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.beans.factory.annotation.Value
import top.bettercode.summer.test.BaseTest

/**
 * @author Peter Wu
 */
@Disabled
internal class AliSmsTemplateTest : BaseTest() {
    @Autowired
    lateinit var aliSmsTemplate: AliSmsTemplate

    @Value("\${mobile}")
    lateinit var mobile: String

    @Disabled
    @Test
    fun sendSms() {
        aliSmsTemplate.sendSms("SMS_126359654", AliSmsReq(mobile, "云图提货", mapOf(
                Pair("code", "123456"))))
    }

    @Test
    fun querySendReport() {
        aliSmsTemplate.querySendReport(mobile)
    }

    @Test
    fun querySmsSignList() {
        aliSmsTemplate.querySmsSignList()
    }

    @Test
    fun querySmsTemplateList() {
        aliSmsTemplate.querySmsTemplateList()
    }
}